inspirehep/modules/workflows/tasks/merging.py

```
@@ -0,0 +1,131 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Tasks related to record merging."""
+
+from __future__ import absolute_import, division, print_function
+
+from invenio_pidstore.resolver import PersistentIdentifier
+
+from inspirehep.utils.record import get_source
+from inspire_json_merger.inspire_json_merger import inspire_json_merge
+
+from ..errors import MissingHeadUUIDError
+from ..utils import (
+    with_debug_logging,
+    retrieve_root_json,
+    retrieve_head_json,
+    store_head_json,
+    store_root_json,
+)
+
+
+@with_debug_logging
+def get_root(obj):
+    """Retrieve the root JSON.
+
+    Retrieves the root JSON corresponding to the current matched record and the
+    provided source.
+    The root version is stored in `obj.extra_data['root_json']`.
+    """
+    source = get_source(obj.data)
+    control_number = obj.data.get('control_number', '')
+    record_uuid = PersistentIdentifier.get('lit', control_number).object_uuid
+    return retrieve_root_json(record_uuid, source)
+
+
+@with_debug_logging
+def get_uuid_from_matched_record(obj):
+    """Retrieve the UUID from the JSON record.
+
+    Retrieves the record UUID .
+    The uuid is stored in `obj.extra_data['head_uuid']`.
+    """
+    recid = obj.extra_data['record_matches']['records'][0]['source']['control_number']
```

This is making quite several assumptions. Shall we raise an appropriate Exception similar to down below? Or maybe it's never going to be necessary.